core/objects.go

```diff
@@ -224,6 +224,22 @@ func (r *Registration) MergeUpdate(input Registration) {
 	}
 }
 
+// ValidationTarget describes a single URL and/or IP addresses resolved for the
+// URL/hostname
+type ValidationTarget struct {
+	URL               string   `json:"url,omitempty"`
+	Hostname          string   `json:"hostname,omitempty"`
+	AddressesResolved []net.IP `json:"addressesResolved"`
+	AddressUsed       net.IP   `json:"addressUsed"`
+}
+
+// ValidationTargets describes all of the URLs and/or IP addresses resolved
+// and used during a single validation attempt
+type ValidationTargets struct {
+	Type string             `json:"type"`
```

Comment:
My thought was to have a separate struct for each validation type, since it may be different for different types. So, e.g. you'd have:

```
type Challenge struct {
  ...
  ValidationRecord *ValidationRecord
}

type ValidationRecord struct {
  SimpleHttp SimpleHttpValidationRecord
  Dvsni DvsiValidationRecord
}

type SimpleHttpValidationRecord []SimpleHttpFetch;

type SimpleHttpFetch struct {
  URL string   `json:"url,omitempty"`
  Hostname string   `json:"hostname,omitempty"`
  AddressesResolved []net.IP `json:"addressesResolved"`
  AddressUsed net.IP   `json:"addressUsed"`
}
```

Also, I suggested AddressesTried because at some point (possibly soon) I expect we will try multiple addresses.